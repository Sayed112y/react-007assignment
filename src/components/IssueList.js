import React from 'react';
import Issue from './Issue';

export default class IssueList extends React.Component {
  render () {

    let issueNodes = this.props.issues && this.props.issues.map(issue => (
        <Issue key={issue.id} id={issue.id} issue_description={issue.issue_description} 
        severity={issue.severity} status={issue.status}/>
    ));

    return (
      <>
        <table border="2">
          <thead>
            <tr className="heading">
              <th>ID</th>
              <th>IssueDescription</th>
              <th>Severity</th>
              <th>Status</th>
            </tr>
          </thead>
          <tbody>
            {issueNodes}
          </tbody>
        </table>
      </>
    );
  }
}
