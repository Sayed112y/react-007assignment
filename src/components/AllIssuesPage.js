import React from "react";
import IssueList from "./IssueList";
import IssueApi from "../data/IssueApi";

export default class AllIssuesPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            issues: []
        };
    }
    componentDidMount() {
        this.setState({ issues: IssueApi.getAllIssues() });
    }
    render() {
        return (
            <>
                <h1>Issue List</h1>
                <IssueList issues={this.state.issues} />
            </>
        );
    }
}
