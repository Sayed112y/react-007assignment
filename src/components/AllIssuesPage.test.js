// Below test case is done with Enzyme’s Full DOM rendering.
//     1. Snapshot testing of AllProductsPage using Eznyme’s Full DOM rendering

// Below test case is done with Enzyme’s Shallow DOM rendering.
//     2. Test case to check if correct heading name is rendered 
//     3. Test case to check if one ProductList React component is rendered

import React from 'react';
import { mount, shallow } from 'enzyme';
import AllIssuesPage from './AllIssuesPage';
import IssueList from './IssueList';
import renderer from 'react-test-renderer'

describe('All Issues Page Snapshot', () => {
    let mountwrapper;
    
    beforeEach(()=>{
        mountwrapper=mount(<AllIssuesPage/>);
    })    

    afterEach(()=>{
        mountwrapper.unmount();
    })

    it('renders correctly', () => {
        expect(mountwrapper).toMatchSnapshot();
    });

});


describe('All Issues Page rendering of elements', () => {
    let wrapper;

    beforeEach(()=>{
        wrapper=shallow(<AllIssuesPage/>);
    })

    afterEach(()=>{
        wrapper.unmount();
    })

    it('renders correct heading for Issues List', () => {
        expect(wrapper.find('h1').render().text()).toEqual("Issue List");
    });

    it('renders one Issue List Component ', () => {
        expect(wrapper.find(IssueList).length).toEqual(1);
    });
});
