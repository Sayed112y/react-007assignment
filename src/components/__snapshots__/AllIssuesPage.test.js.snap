// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Issues Page Snapshot renders correctly 1`] = `
<AllIssuesPage>
  <h1>
    Issue List
  </h1>
  <IssueList
    issues={
      Array [
        Object {
          "id": 1,
          "issue_description": "On clicking Delete,the applicationcrashes",
          "severity": "Critical",
          "status": "Open",
        },
        Object {
          "id": 2,
          "issue_description": "The heading Add is wrongly displayed as edit",
          "severity": "Minor",
          "status": "Closed",
        },
        Object {
          "id": 3,
          "issue_description": "The payment functionality is missing",
          "severity": "Major",
          "status": "missing",
        },
      ]
    }
  >
    <table
      border="2"
    >
      <thead>
        <tr
          className="heading"
        >
          <th>
            ID
          </th>
          <th>
            Issue Description
          </th>
          <th>
            SEVERITY
          </th>
          <th>
            STATUS
          </th>
        </tr>
      </thead>
      <tbody>
        <Issue
          id={1}
          issue_description="On clicking Delete,the applicationcrashes"
          key="1"
          severity="Critical"
          status="Open"
        >
          <tr>
            <td>
              1
            </td>
            <td>
              On clicking Delete,the applicationcrashes
            </td>
            <td>
              Critical
            </td>
            <td>
              Open
            </td>
          </tr>
        </Issue>
        <Issue
          id={2}
          issue_description="The heading Add is wrongly displayed as edit"
          key="2"
          severity="Minor"
          status="Closed">
          <tr>
            <td>
              2
            </td>
            <td>
              The payment functionality is missing            </td>
            <td>
              Minor
            </td>
            <td>
              Closed
            </td>
          </tr>
        </Issue>
        <Issue
          id={3}
          issue_description="The payment functionality is missing"
          key="3"
          severity="Major"
          status="missing"
        >
          <tr>
            <td>
              3
            </td>
            <td>
              The payment functionality is missing
            </td>
            <td>
              Major
            </td>
            <td>
              missing
            </td>
          </tr>
        </Issue>
      </tbody>
    </table>
  </IssueList>
</AllIssuesPage>
`;
